tests/common.go

```diff
@@ -5314,3 +5314,88 @@ func GetSubsetOfSlice[T any](items []T, length int) ([]T, error) {
 	}
 	return randomItems, nil
 }
+
+func DeleteGivenPoolInNode(stNode node.Node, poolIDToDelete string) (err error) {
+
+	log.InfoD("Setting pools in maintenance on node %s", stNode.Name)
+	if err = Inst().V.EnterPoolMaintenance(stNode); err != nil {
+		return err
+	}
+	//Waiting for cli to work
+	time.Sleep(2 * time.Minute)
+
+	if status, err := Inst().V.GetNodeStatus(stNode); err != nil {
+		return err
+	} else {
+		log.InfoD("Node [%s] has status: [%v] after entering pool maintenance", stNode.Name, status)
+	}
+
+	expectedStatus := "In Maintenance"
+	if err = waitForPoolStatusToUpdate(stNode, expectedStatus); err != nil {
+		return fmt.Errorf("node %s pools are not in status %s. Error:%v", stNode.Name, expectedStatus, err)
+	}
+	defer func() {
+		var exitErr error
+		if exitErr = Inst().V.ExitPoolMaintenance(stNode); exitErr != nil {
+			log.Errorf("error exiting pool maintenance in the node [%v]. Err: %v", stNode.Name, exitErr)
+			return
```

Comment:
Let's not `log.Errorf()` and just return verbose error

Reply:
this logic is in defer function, where if I return error will mask the test error if any , since this is just a clean up logic, Im just printing error when any error occurred.